src/templates/middleware.js

```diff
@@ -10,12 +10,18 @@ middleware.nuxti18n = async (context) => {
   }
 
   // Handle root path redirect
+  const rootRedirect = <%= JSON.stringify(options.rootRedirect) %>
   if (route.path === '/' && rootRedirect) {
+    if (rootRedirect.path && rootRedirect.statusCode) {
+      redirect(rootRedirect.statusCode, '/' + rootRedirect.path, route.query)
+      return
+    }
+
```

Comment:
Maybe refactor so that you don't have to repeat `redirect` call twice:

```js
let statusCode = 302
let path = rootRedirect

if (typeof rootRedirect !== 'string') {
  { statusCode, path } = rootRedirect
}

redirect(statusCode, '/' + path, route.query)
```